thrust/system/detail/sequential/scan.h

```diff
@@ -54,6 +54,8 @@ __host__ __device__
 
   // Use the input iterator's value type per https://wg21.link/P0571
   using ValueType = typename thrust::iterator_value<InputIterator>::type;
+  // Use for explicit type conversion
+  using OutputType = typename thrust::iterator_value<OutputIterator>::type;
```

Comment:
This was my thinking, that the output type would make sense, but when I do include this I get compile time errors

Reply:
Output iterators are not required to have a value type.

Reply:
@ericniebler Thanks, for those that dont have it is it set to null?

Reply:
Thanks for the tips, I will look into implementing that. There may still be times where when it has to pick the Fallback type (which I am sensing would be ValueType to not break anything), there may still be warning.